crates/rpc/src/handlers/light_client.rs

```diff
@@ -41,3 +45,96 @@ pub async fn get_light_client_bootstrap(
 
     Ok(HttpResponse::Ok().json(DataVersionedResponse::new(light_client_bootstrap)))
 }
+
+#[get("/beacon/light_client/updates")]
+pub async fn get_light_client_updates(
+    db: Data<ReamDB>,
+    start_period: Query<u64>,
+    count: Query<u64>,
+) -> Result<impl Responder, ApiError> {
+    let start_period: u64 = start_period.into_inner();
+    let count = std::cmp::min(count.into_inner(), MAX_REQUEST_LIGHT_CLIENT_UPDATES);
+
+    let mut updates = Vec::new();
+
+    for period in start_period..start_period + count {
+        let slot = period * EPOCHS_PER_SYNC_COMMITTEE_PERIOD * SLOTS_PER_EPOCH;
+        let block_root = db
+            .slot_index_provider()
+            .get(slot)
+            .map_err(|err| {
+                error!("Failed to get block root for slot {}: {:?}", slot, err);
+                ApiError::InternalError
+            })?
+            .ok_or_else(|| {
+                ApiError::NotFound(format!(
+                    "No block root found for slot {} (period {})",
+                    slot, start_period
+                ))
+            })?;
+
+        let block = db
+            .beacon_block_provider()
+            .get(block_root)
+            .map_err(|_| ApiError::InternalError)?
+            .ok_or(ApiError::NotFound(format!(
+                "Failed to find beacon_block from {block_root:?}"
+            )))?;
+
+        let state = db
+            .beacon_state_provider()
+            .get(block_root)
+            .map_err(|_| ApiError::InternalError)?
+            .ok_or(ApiError::NotFound(format!(
+                "Failed to find beacon_state from {block_root:?}"
+            )))?;
+
+        let attested_block = db
+            .beacon_block_provider()
+            .get(block.message.parent_root)
+            .map_err(|_| ApiError::InternalError)?
```

Comment:
```suggestion
            .map_err(|err| ApiError::InternalError(format!("XYZ: {err:?}"))?
```

you missed here and a few other places, could you make InternalError take a String instead of logging and error